modules/standard/Spawn.chpl

```diff
@@ -711,60 +721,81 @@ module Spawn {
         to its output while the parent process is waiting for it to complete
         (and not consuming its output).
 
-    :arg error: optional argument to capture any error encountered
-                when waiting for the child process.
     :arg buffer: if `true`, buffer input and output pipes (see above).
-
    */
-  proc subprocess.wait(out error:syserr, buffer=true) {
+  proc subprocess.wait(buffer=true) throws {
+    try _throw_on_launch_error();
 
     if buffer {
-      this.communicate(error);
+      try this.communicate();
       return;
     }
 
-    error = this.spawn_error;
-    if !running then return;
+    var err: syserr = this.spawn_error;
+    if !running {
+      if err then try ioerror(err, "in subprocess.wait");
+      return;
+    }
 
+    // TODO: rethink this approach, perhaps with an Error list, or var Error
+    // - complete all these throwing calls, even if they throw
+    //   - use catch { } to continue despite an Error
+    //   - extract a syserr if a SystemError is caught
+    // - at the end, throw an error if err is set
     on home {
-
       // Close stdin.
       if this.stdin_pipe {
         // send data to stdin
         _stop_stdin_buffering();
-        this.stdin_channel.close(error=error);
+        try {
+          this.stdin_channel.close();
+        } catch e: SystemError {
+          err = e.err;
+        } catch { }
       }
 
       // wait for child process to terminate
       var done:c_int = 0;
       var exitcode:c_int = 0;
-      error = qio_waitpid(pid, 1, done, exitcode);
+      err = qio_waitpid(pid, 1, done, exitcode);
```

Comment:
This function had this problem before you got to it, but we should be using different error variables for the different operations, initializing them each with 0, and then combining them at the end, so that if any one operation fails we get an error out.

Reply:
I tried having each error case check if `err` was already set, with the `qio_waitpid` error taking precedence. Let me know if this is what you had in mind.

Reply:
Alright I gave it another shot. If it looks good to you I'll fire off testing.